src/presentation/http/router/note.test.ts

```diff
@@ -50,48 +53,83 @@ describe('Note API', () => {
-    test('Returns note with access rights by public id with 200 status when note is publicly available', async () => {
-      const user = await global.db.insertUser();
+    test.each([
+      /** Returns 200 if user is team member with a Read role */
+      {
+        role: MemberRole.Read,
+        isPublic: false,
+        isAuthorized: true,
+        expectedStatusCode: 200,
+      },
 
-      const note = await global.db.insertNote({
-        creatorId: user.id,
-      });
+      /** Returns 200 if user is team member with a Write role */
+      {
+        role: MemberRole.Write,
+        isPublic: false,
+        isAuthorized: true,
+        expectedStatusCode: 200,
+      },
 
-      await global.db.insertNoteSetting({
-        noteId: note.id,
+      /** Returns 200 if note is public */
+      {
+        role: null,
         isPublic: true,
-      });
+        isAuthorized: false,
+        expectedStatusCode: 200,
+      },
 
-      const response = await global.api?.fakeRequest({
-        method: 'GET',
-        url: `/note/${note.publicId}`,
-      });
+      /** Returns 403 if user is not in the team */
+      {
+        role: null,
+        isPublic: false,
+        isAuthorized: true,
+        expectedStatusCode: 403,
+      },
 
-      expect(response?.statusCode).toBe(200);
+      /** Returns 403 if user is not authorized */
+      {
+        role: null,
+        isPublic: false,
+        isAuthorized: false,
+        expectedStatusCode: 403,
+      },
+    ])
+    ('Returns note with access rights by public id', async ({ role, isPublic, isAuthorized, expectedStatusCode }) => {
+      /** Only if user has a Write role, he can edit the note */
+      const canEdit = role === MemberRole.Write;
 
-      expect(response?.json()).toMatchObject({
-        'note': {
-          'id': note.publicId,
-        },
-        'accessRights': {
-          'canEdit': false,
-        },
-      });
-    });
+      /** Create test user - creator of note */
+      const creator = await global.db.insertUser();
 
-    test('Returns note by public id with 200 status when access is disabled, but user is in team', async () => {
+      /** Create test user */
       const user = await global.db.insertUser();
 
+      /** Create test note */
       const note = await global.db.insertNote({
-        creatorId: user.id,
+        creatorId: creator.id,
       });
 
+      /** Create test note settings */
       await global.db.insertNoteSetting({
         noteId: note.id,
-        isPublic: false,
+        isPublic: isPublic,
       });
 
-      const accessToken = global.auth(user.id);
+      /** Create test team */
+      if (role !== null) {
+        await global.db.insertNoteTeam({
+          noteId: note.id,
+          userId: user.id,
+          role: role,
+        });
+      }
+
+      /** If user is not authorized, accessToken is empty */
+      let accessToken = '';
+
+      if (isAuthorized) {
+        accessToken = global.auth(user.id);
+      }
```

Comment:
Then the variable will be out of scope, it must be created outside if ()

Reply:
![image](https://github.com/codex-team/notes.api/assets/59017579/fe3ee511-5c6a-4c78-96a0-a1d063aa367b)
Then it won't be seen outside of scope (if user is not Authorized) ^^^

I think it's better my way, where we provide an empty token instead of having another if else around the reply request.